crates/bevy_pbr/src/render/mesh.rs

```diff
@@ -106,81 +107,52 @@ bitflags::bitflags! {
     }
 }
 
+pub type ExtractMeshQuery = (
+    Entity,
+    &'static ComputedVisibility,
+    &'static GlobalTransform,
+    &'static Handle<Mesh>,
+    Option<&'static NotShadowReceiver>,
+);
+type ExtractMeshItem<'w> = QueryItem<'w, ExtractMeshQuery>;
+
 pub fn extract_meshes(
     mut commands: Commands,
-    mut previous_caster_len: Local<usize>,
-    mut previous_not_caster_len: Local<usize>,
-    caster_query: Query<
-        (
-            Entity,
-            &ComputedVisibility,
-            &GlobalTransform,
-            &Handle<Mesh>,
-            Option<&NotShadowReceiver>,
-        ),
-        Without<NotShadowCaster>,
-    >,
-    not_caster_query: Query<
-        (
-            Entity,
-            &ComputedVisibility,
-            &GlobalTransform,
-            &Handle<Mesh>,
-            Option<&NotShadowReceiver>,
-        ),
-        With<NotShadowCaster>,
-    >,
+    mut prev_len_caster: Local<usize>,
+    mut prev_len_not: Local<usize>,
+    caster_query: Query<ExtractMeshQuery, Without<NotShadowCaster>>,
```

Comment:
I really don't know about the perf characteristics. But a better way to do this would be to merge the two queries and replace the `Option<&'static NotShadowReceiver>` by `With<NotShadowReceiver>` (the negative on the marker type really throws me off)